src/views/Checkout.vue

```diff
@@ -51,28 +221,148 @@ export default class Checkout extends Vue {
         client.import(request).catch(console.error); // TODO: proper error handling
     }
 
-    private async created() {
+    @Emit()
+    private close() {
+        this.$rpc.reject(new Error('CANCEL'));
+    }
+
+    private get processedWallets(): WalletInfo[] {
+        const singleAccounts = new Map<string, AccountInfo>();
+
+        const filteredWallets = this.wallets.filter((wallet) => {
+            if (wallet.type !== WalletType.LEGACY) return true;
+
+            const accountArray = Array.from(wallet.accounts.entries())[0];
+            accountArray[1].walletId = wallet.id;
+            singleAccounts.set(accountArray[0], accountArray[1]);
+            return false;
+        });
+
+        if (singleAccounts.size > 0) {
+            filteredWallets.push(new WalletInfo(
+                LEGACY_GROUPING_WALLET_ID,
+                LEGACY_GROUPING_WALLET_LABEL,
+                singleAccounts,
+                [],
+                WalletType.LEGACY,
+            ));
+        }
+
+        return filteredWallets;
+    }
+
+    private get hasSufficientBalanceAccount(): boolean {
+        const minBalance = this.request.value + (this.request.fee || 0);
+        return this.wallets.find((wallet: WalletInfo) => {
+            return Array.from(wallet.accounts.values()).find((account: AccountInfo) => {
+                return account.balance ? account.balance >= minBalance : false;
+            }) !== undefined;
+        }) !== undefined;
+    }
+
+    private async handleOnboardingResult() {
+        // Check if we are returning from an onboarding request
         if (staticStore.sideResult && !(staticStore.sideResult instanceof Error)) {
             const sideResult = staticStore.sideResult as LoginResult;
-            // Add imported wallet to store and pre-select wallet (if regular wallet)
-            // or stay on first page for imported legacy wallet
 
+            // Add imported wallet to Vuex store
             const walletInfo = await WalletStore.Instance.get(sideResult.walletId);
-            if (!walletInfo) return;
-            this.$addWallet(walletInfo);
+            if (walletInfo) {
+                this.$addWallet(walletInfo);
+                this.sideResultAddedWallet = true;
 
-            // Set as activeWallet and activeAccount
-            this.$setActiveAccount({
-                walletId: walletInfo.id,
-                userFriendlyAddress: walletInfo.accounts.values().next().value.userFriendlyAddress,
-            });
+                // Set as activeWallet and activeAccount
+                this.$setActiveAccount({
+                    walletId: walletInfo.id,
+                    userFriendlyAddress: walletInfo.accounts.values().next().value.userFriendlyAddress,
+                });
+            }
         }
         delete staticStore.sideResult;
     }
 
-    @Emit()
-    private close() {
-        this.$rpc.reject(new Error('CANCEL'));
+    private getCache() {
```

Comment:
The semantics of this cache is not to store the last known blockchain height (as it is not updated in `onHeadChange` which is on purpose), therefore the name `cache` is misleading.
I would call it `getLastBalanceCheckTime` or something like this.